app/main.py

```diff
@@ -2,5 +1,9 @@
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    def wrapper(*args, **kwargs) -> Any:
+        result = func(*args, **kwargs)
+        return result
+    return wrapper
```

Comment:
The `cache` decorator should handle multiple functions simultaneously, ensuring that cached results are managed separately for each function. This aspect is not addressed in the current implementation.